include/boost/url/detail/impl/any_segments_iter.ipp

```diff
@@ -201,6 +206,7 @@ segment_iter(
     : any_segments_iter(s_)
 {
     front = s;
+    fast_nseg = 1;
```

Comment:
Even for empty strings?

Reply:
Yes, because `segment_iter` is for exactly one segment (see where it is used)